tests/transaction_tests.cpp

```diff
@@ -102,20 +102,60 @@ TEST_CASE("Transaction guard") {
 
     SECTION("insert, call make a storage to call an exception and check that rollback was fired") {
         auto countBefore = storage.count<Object>();
-        try {
-            auto guard = storage.transaction_guard();
+        SECTION("transaction_guard") {
+            try {
```

Comment:
Can we use a dedicated Catch2 test macro for this, please?
```c++
REQUIRE_THROWS_WITH(storage.transaction([&] {
    storage.insert(Object{0, "John"});
    storage.get<Object>(0);
    return true;
}),
                    "Not found");
```

Reply:
let me try

Reply:
actually I'd like to test exception's error code more than its description string. But I don't see any option of doing it with `REQUIRE_THROWS_WITH` so I'd like to keep the things as they are

Reply:
done. Please check